### PR TITLE
[libwebm] apply upstream changes to support Android NDK r15b and later

### PR DESCRIPTION
**Pull request by jwtowner**
This applies a minimal set of upstream changes from the master branch of libwebm to 1.0.0.27 in order to support building with Android NDK r15b and later up through Android NDK r23 beta. This was primarily tested with Android NDK r21e and Android NDK r22.

Commits taken from https://chromium.googlesource.com/webm/libwebm master branch:

[0ae757087f5e6eb01dfea16cc09205b2425cfb74](https://chromium.googlesource.com/webm/libwebm/+/0ae757087f5e6eb01dfea16cc09205b2425cfb74)
Fix android build failure with NDK r15b.

[90967863b2f67962f52595a26abb8e4fa4105d44](https://chromium.googlesource.com/webm/libwebm/+/90967863b2f67962f52595a26abb8e4fa4105d44)
mkvparser: fix float conversion warning

EDIT: This PR allows building libwebm for the android community triplets.

### CODE REVIEW

File: ports/libwebm/CONTROL [hunk 1]
```diff
@@ -1,4 +1,5 @@
 Source: libwebm
 Version: 1.0.0.27-5
+Port-Version: 1
```
Comment (NancyLi1013) [inline ports/libwebm/CONTROL:3]:
```suggestion
Version: 1.0.0.27
Port-Version: 6
```

File: versions/baseline.json [hunk 2]
```diff
@@ -3514,7 +3514,7 @@
     },
     "libwebm": {
       "baseline": "1.0.0.27-5",
```
Comment (NancyLi1013) [inline versions/baseline.json:3516]:
```suggestion
      "baseline": "1.0.0.27",
```

File: versions/baseline.json [hunk 3]
```diff
@@ -3514,7 +3514,7 @@
...
-      "port-version": 0
+      "port-version": 1
```
Comment (NancyLi1013) [inline versions/baseline.json:3517]:
```suggestion
      "port-version": 6
```

File: versions/l-/libwebm.json [hunk 4]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "52d2a31fc4b3cb384cdac69a15b50f8d8a6e3082",
+      "version-string": "1.0.0.27-5",
+      "port-version": 1
```
Comment (NancyLi1013) [inline versions/l-/libwebm.json:6]:
```suggestion
      "version-string": "1.0.0.27",
      "port-version": 6
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]
Could you please update 
```
file(COPY ${SOURCE_PATH}/LICENSE.TXT DESTINATION ${CURRENT_PACKAGES_DIR}/share/libwebm)
file(RENAME ${CURRENT_PACKAGES_DIR}/share/libwebm/LICENSE.TXT ${CURRENT_PACKAGES_DIR}/share/libwebm/copyright)
```
as

`file(INSTALL ${SOURCE_PATH}/LICENSE.TXT DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`?

**Comment by jwtowner** [comment]
@NancyLi1013 

Done! I've switched the portfile over to use `file(INSTALL ... RENAME ...)` when installing the copyright file as you've suggested.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by Hoikas** [comment]
Considering it's been a little over five years since the last release and that these patches came directly from upstream master, maybe it would be better to change this port to a live-at-head policy. Meaning, use a date based version-string and use the current HEAD in the portfile.

**Comment by jwtowner** [comment]
> Considering it's been a little over five years since the last release and that these patches came directly from upstream master, maybe it would be better to change this port to a live-at-head policy. Meaning, use a date based version-string and use the current HEAD in the portfile.

Yeah, I agree, and I would be pleased to do the work for that. However, I'd like to do it after getting this PR merged if that's okay, so we at least have libwebm 1.0.0.27 to fallback to in case we run into other problems on other platforms and for those using port versioning. With this PR, I at least have it working on Windows, UWP, Android, Linux, iOS and macOS targets at the current version.

**Comment by ras0219-msft** [comment]
Awesome PR, thanks!

**Pull request merged**